src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1377,91 +1376,55 @@ async def import_pool(self, job, data):
                 }]
             }
         """
-
-        pool = None
-        for p in await self.middleware.call('zfs.pool.find_import'):
-            if p['guid'] == data['guid']:
-                pool = p
-                break
-        if pool is None:
-            raise CallError(f'Pool with guid "{data["guid"]}" not found', errno.ENOENT)
-
+        # import zpool
         try:
-            job.check_pipe("input")
-            key = job.pipes.input.r
-        except ValueError:
-            key = None
-
-        passfile = None
-        if key and data.get('passphrase'):
-            encrypt = 2
-            passfile = tempfile.mktemp(dir='/tmp/')
-            with open(passfile, 'w') as f:
-                os.chmod(passfile, 0o600)
-                f.write(data['passphrase'])
-        elif key:
-            encrypt = 1
-        else:
-            encrypt = 0
-
-        pool_name = data.get('name') or pool['name']
-        pool_id = None
-        try:
-            pool_id = await self.middleware.call('datastore.insert', 'storage.volume', {
-                'vol_name': pool_name,
-                'vol_encrypt': encrypt,
-                'vol_guid': data['guid'],
-                'vol_encryptkey': str(uuid.uuid4()) if encrypt else '',
-            })
-            pool = await self.middleware.call('pool.query', [('id', '=', pool_id)], {'get': True})
-            if encrypt > 0:
-                if not os.path.exists(GELI_KEYPATH):
-                    os.mkdir(GELI_KEYPATH)
-                with open(pool['encryptkey_path'], 'wb') as f:
-                    f.write(key.read())
-
-            await self.middleware.call('pool.scrub.create', {
-                'pool': pool_id,
-            })
-
-            await self.middleware.call('zfs.pool.import_pool', pool['guid'], {
-                'altroot': '/mnt',
-                'cachefile': ZPOOL_CACHE_FILE,
-            }, True, None, pool_name)
-
-            await self.middleware.call('zfs.dataset.update', pool_name, {
-                'properties': {
-                    'aclinherit': {'value': 'passthrough'},
-                },
-            })
+            guid = data['guid']
+            opts = {'altroot': '/mnt', 'cachefile': ZPOOL_CACHE_FILE}
+            any_host = True
+            use_cachefile = None
+            new_name = data.get('name')
```

Comment:
Btw, let's please add validation to ensure that the pool name is not already in use

Reply:
That's already validated by the import process?

Reply:
The error raised in that case would be `Failed to import pool with guid 1234` :)

Reply:
Then the error in `py-libzfs` should be fixed.

Reply:
Could you please clarify which one we are referring to ? Also we catch the exception and we do not propagate it appropriately even with better handling in the underlying zfs plugin or py-libzfs ?

Reply:
Sure, you asked to check to make sure the pool doesn't already exist. (Meaning it's already imported)
```
>>> c.call('zfs.pool.import_pool', 'ha2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/middlewared/client/client.py", line 465, in call
    raise ClientException(c.error, c.errno, c.trace, c.extra)
middlewared.client.client.ClientException: [ENOENT] Pool ha2 not found.
>>>
root[~]# zpool list
NAME        SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
boot-pool    31G  2.73G  28.3G        -         -     0%     8%  1.00x    ONLINE  -
ha2        7.50G  1.36M  7.50G        -     1.50G     0%     0%  1.00x    ONLINE  /mnt
tank       7.50G  1.33G  6.17G        -     1.50G     7%    17%  1.00x    ONLINE  /mnt
root@[~]# zpool import ha2
cannot import 'ha2': a pool with that name already exists
```

Why is `zfs.pool.import_pool` not raising `EEXIST`?